### PR TITLE
[#3263] Add idle timeout.

### PR DESCRIPTION
**Pull request by suzanaAngelovaIvanova**
The idle timeout is used when data is not received, the connection is timeout and closed.

Signed-off-by: Ivanova Suzana <Suzana.Ivanova@bosch.io>

### CODE REVIEW

File: core/src/main/java/org/eclipse/hono/config/ServerConfig.java [hunk 1]
```diff
@@ -275,4 +277,36 @@ public final void setSni(final boolean sni) {
...
+     * Gets the idle timeout.
+     * <p>
+     * A connection will timeout and be closed if no data is received or sent within the idle timeout period.
+     * A zero value means no timeout is used.
+     * <p>
+     * The default value is {@code 60} in seconds.
+     *
+     * @return The idle timeout in seconds.
+     */
+    public int getIdleTimeout() {
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/config/ServerConfig.java:291]: can you please make this final?

File: core/src/main/java/org/eclipse/hono/config/ServerConfig.java [hunk 2]
```diff
@@ -275,4 +277,36 @@ public final void setSni(final boolean sni) {
...
+     * <p>
+     * A connection will timeout and be closed if no data is received or sent within the idle timeout period.
+     * A zero value means no timeout is used.
+     * <p>
+     * The default value is {@code 60}. The idle timeout is in seconds.
+     *
+     * @param idleTimeout The idle timeout.
+     * @throws IllegalArgumentException if idleTimeout is less than {@code 0}.
+     */
+    public void setIdleTimeout(final int idleTimeout) {
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/config/ServerConfig.java:306]: can you please make this final?

File: core/src/test/resources/server-options.yaml [hunk 3]
```diff
@@ -8,3 +8,4 @@ hono:
     nativeTlsRequired: true
     port: 11000
     trustStoreFormat: "JKS"
+    idleTimeout: 60
```
Comment (sophokles73) [inline core/src/test/resources/server-options.yaml:11]: missing EOL

File: site/documentation/content/admin-guide/http-adapter-config.md [hunk 4]
```diff
@@ -27,7 +27,8 @@ configuring the HTTP adapter.
...
-| `HONO_HTTP_DEFAULTSENABLED`<br>`hono.http.defaultsEnabled` | no | `true` | If set to `true` the protocol adapter uses *default values* registered for a device and/or its tenant to augment messages published by the device with missing information like a content type. In particular, the protocol adapter adds such default values as Kafka record headers or AMQP 1.0 message (application) properties before the message is sent downstream. |
+| `HONO_HTTP_DEFAULTSENABLED`<br>`hono.http.defaultsEnabled` | no | `true` | If set to `true` the protocol adapter uses *default values* registered for a device to augment messages published by the device with missing information like a content type. In particular, the protocol adapter adds default values registered for the device as (application) properties with the same name to the AMQP 1.0 messages it sends downstream to the AMQP Messaging Network. |
```
Comment (sophokles73) [inline site/documentation/content/admin-guide/http-adapter-config.md:30]: Did you change the description here on purpose?

**Review by sophokles73 — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/eclipse/hono/config/ServerConfig.java [hunk 5]
```diff
@@ -275,4 +277,36 @@ public final void setSni(final boolean sni) {
...
+     * Gets the idle timeout.
+     * <p>
+     * A connection will timeout and be closed if no data is received or sent within the idle timeout period.
+     * A zero value means no timeout is used.
+     * <p>
+     * The default value is {@code 60} in seconds.
+     *
+     * @return The idle timeout in seconds.
+     */
+    public final int getIdleTimeout() {
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/config/ServerConfig.java:291]: Looks like this class and the *AmqpAdapterProperties* classes have a different understanding of the semantics of this property. First, this class returns #seconds while *AmqpAdapterProperties.getIdleTimeout()* seems to return #millis. Second, in *AmqpAdapterProperties* the *idleTimeout* is used to keep an AMQP connection to a device alive whereas in this class, the property is supposed to be used for terminating a device's HTTP request.
FMPOV we need to keep the *AmqAdapterProperties* unchanged and instead should move the *idleTimeout* property from this class to both *HttpProtocolAdapterProperties* and *HttpServiceConfigProperties* ...

@calohmn WDYT?

Comment (calohmn) [reply core/src/main/java/org/eclipse/hono/config/ServerConfig.java:291]: Oh, I missed that. 
> FMPOV we need to keep the *AmqAdapterProperties* unchanged and instead should move the *idleTimeout* property from this class to both *HttpProtocolAdapterProperties* and *HttpServiceConfigProperties* ...

I agree.

Comment (suzanaAngelovaIvanova) [reply core/src/main/java/org/eclipse/hono/config/ServerConfig.java:291]: Ok. I'll change it :)

**Comment by calohmn** [comment]
@suzanaAngelovaIvanova Can you create another PR to apply the changes made in response to the comments here on master?

**Pull request merged**